Extensions/PrimitiveDrawing/Extension.cpp

```diff
@@ -632,5 +633,162 @@ void DeclarePrimitiveDrawingExtension(gd::PlatformExtension& extension) {
       .SetFunctionName("AreCoordinatesRelative")
       .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
 
+  obj.AddAction("Scale",
+                _("Scale"),
+                _("Modify the scale of the specified object."),
+                _("the scale"),
+                _("Size"),
+                "res/actions/scale24.png",
+                "res/actions/scale.png")
+      .AddParameter("object", _("Object"), "Drawer")
+      .UseStandardOperatorParameters("number")
+      .MarkAsAdvanced();
+
+  obj.AddExpressionAndConditionAndAction("number",
+                "ScaleX",
+                _("Scale on X axis"),
+                _("the width's scale of an object"),
+                _("the width's scale"),
+                _("Size"),
+                "res/actions/scaleWidth24.png")
+      .AddParameter("object", _("Object"), "Drawer")
+      .UseStandardParameters("number")
+      .MarkAsAdvanced();
```

Comment:
I should probably revert the use of `AddExpressionAndConditionAndAction` to keep the same name as Sprite to make a generalization easier.

Reply:
I'm not sure... if we generalize this later, the full name of the actions (PrimitiveDrawing::ScaleX) will be different anyway. We'll probably move these to the base RuntimeObject and "redirect" any existing action to the base RuntimeObject methods.

What will be important will be to easily find all the actions and ensure we've followed the exact same parameters. In this sense, AddExpressionAndConditionAndAction actually helps because it's less code to update in the future :)